lib/checks/navigation/identical-links-same-purpose.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Note:
+ * `identical-links-same-purpose-after` fn, helps reconcile the results & alter the CheckResult accordingly
+ */
+const { text } = axe.commons;
+const accessibleText = text.accessibleTextVirtual(virtualNode);
+const linkResource = getLinkResource(node);
+
+if (!linkResource) {
+	return undefined;
+}
+
+/**
+ * Set `data` for use in `after` fn
+ */
+this.data({
+	accessibleText,
+	linkResource
+});
+
+return true;
+
+/**
+ * Get resource that a element points to (eg: href or location redirects)
+ * @param {HTMLElement} el Element
+ * @returns {String|null}
+ */
+function getLinkResource(el) {
+	if (el.hasAttribute('href')) {
```

Comment:
use `elm.href`, rather than `getAttribute('href')`. It'll give you the complete URL. That way hrefs that are different, but resolve to the same page are considered the same.